docs/docs/platform/quickstarts/redwoodjs.mdx

```diff
@@ -0,0 +1,1178 @@
+user's ID and the `X-Hasura-User-ID` session variable, which is passed with each
+requests.
+
+![Hasura users permissions](/img/quickstarts/hasura-permissions-2.png)
+
+Next, select the columns you'd like the users to have access to, and click
+**Save Permissions**.
+
+![Hasura users permissions](/img/quickstarts/hasura-permissions-3.png)
+
+Repeat the same steps on the `update` operation for the `user` role to allow
+users to update their `displayName` and `metadata` only.
+
+Finally, to add caching, synchronizing, and updating server state in your RedwoodJS app, let's refactor the user data fetching using the Apollo client and our GraphQL API instead.
+
+So, first, add the following GraphQL query to retrieve the current user data, and use React Context to run that query and share its result throughout your application:
+
+```js title="web/src/UserProvider.js"
+import React, { useContext } from 'react'
+import { useAuth } from '@redwoodjs/auth'
+import { useQuery } from '@redwoodjs/web'
+
+const GET_USER_QUERY = gql`
+  query GetUser($id: uuid!) {
+    user(id: $id) {
+      id
+      email
+      displayName
+      metadata
+      avatarUrl
+    }
+  }
+`
+
+const UserContext = React.createContext(null)
+
+export function UserProvider({ children = null }) {
+  const { userMetadata } = useAuth()
+
+  const { loading, error, data } = useQuery(GET_USER_QUERY, {
+    variables: { id: userMetadata?.id }
+  })
+  const user = data?.user
+
+  if (error) {
+    return <p>Something went wrong. Try to refresh the page.</p>
+  }
+  if (loading) {
+    return null
+  }
+
+  return <UserContext.Provider value={{ user }}>{children}</UserContext.Provider>
+}
+
+export function useUserContext() {
+  return useContext(UserContext)
+}
+```
+
+Then, add the `UserProvider` inside your `Routes.js` file as follow:
+
+```jsx title="web/src/Routes.js"
+// highlight-next-line
+import { UserProvider } from './UserProvider'
+
+const Routes = () => {
+  return (
+    <Router>
+      <Private unauthenticated="signIn">
+        {/* highlight-next-line */}
+        <Set wrap={[UserProvider, DashLayout]}>
+          <Route path="/" page={HomePage} name="home" />
+          <Route path="/profile" page={ProfilePage} name="profile" />
+        </Set>
+      </Private>
+      <Route path="/sign-up" page={SignUpPage} name="signUp" />
+      <Route path="/sign-in" page={SignInPage} name="signIn" />
+      <Route notfound page={NotFoundPage} />
+    </Router>
+  )
+}
+```
+
+Finally, replace the `userMetadata` with the `useUserContext` hook to retrieve the current user data inside the following files:
+
+<Tabs groupId="files">
+  <TabItem value="DashLayout" label="DashLayout" default>
+
+```js title="web/src/layouts/DashLayout/DashLayout.js"
+// highlight-next-line
+import { useUserContext } from 'src/UserProvider'
+
+const DashLayout = ({ children }) => {
+  // highlight-next-line
+  const { user } = useUserContext()
```

Comment:
Since this code goes from:

```js
const { logOut, userMetadata: user } = useAuth()
```

To:

```js
const { user } = useUserContext()
const { logOut } = useAuth()
```

I would include the second line with `const { logOut } = useAuth()` to avoid confusion.

Reply:
✅ Updated.